dts/arm/st/f7/stm32f7.dtsi

```diff
@@ -638,7 +638,6 @@
 			compatible = "st,stm32-rng";
 			reg = <0x50060800 0x400>;
 			clocks = <&rcc STM32_CLOCK_BUS_AHB2 0x00000040>;
-			status = "disabled";
```

Comment:
I feel that this "status" field should be removed from other ST SoC device tree files as well.. The "status" field is primarily valuable for cases where there are multiple instances of the same device class and we want to enable a subset (given that the driver is built into the zephyr kernel) of these devices.. For example, the SoC may support 6 instances of I2C and we want to enable only 2.. 

However, for the RNG, there is generally only a single instance of the Random Number Generator in the hardware, and so the value of keeping the "status" field goes away. Hence the removal.. I feel that this should be removed from other ST SoC Device Tree files as well.. Thoughts ?

Reply:
> The "status" field is primarily valuable for cases where there are multiple instances of the same device and we want to enable a subset

Even when there is only single instance of a device, `status` field can be used to specify whether the associated hardware peripheral is enabled, so that rationale is not valid.

IMO, consistently specifying `status` field is better.

Reply:
@stephanosio , there are places where people have decided to omit the "status" field because it adds little or no value. For example, the zynqmp dtsi file does not specify the status field for PMU Firmware nodes as seen here: https://github.com/torvalds/linux/blob/master/arch/arm64/boot/dts/xilinx/zynqmp.dtsi#L106

This implicitly means that the device status should be considered "okay" (now part of the device tree specifications as well: https://github.com/devicetree-org/devicetree-specification/commit/fcf50a07064ec1721bf5315a27df3b9778dee23a), and it is up-to the kernel to use this node or not. If the kernel driver for this device is built, it will use the device and ignore otherwise.

I have no problem in reverting to add the "status" back but in such a case, I would expect the kernel driver to use this property to boot up the device..

Reply:
While it is true that lack of `status` property is interpreted as "okay" (or enabled), I am not sure if this implicit `status` is a convention widely used in Zephyr (as far as I can see, we are verbosely specifying the `status` almost everywhere).

If we are to start enforcing this convention, opening an RFC and documenting it first would be great; otherwise, it is foreseeable that we will have another commit in the future that says "add missing `status` properties for X."

> I would expect the kernel driver to use this property to boot up the device (which its not doing right now)..

Although we may not be using the `status` property in many drivers right now, that does not necessarily have to be case in the future (as you may know, migration to device tree is largely incomplete/only partial for the majority of platforms at this time).

Personally, given that, unlike Linux, we are often targeting ultra low power embedded devices, I can still appreciate the verbosity here.

Reply:
If Zephyr is not using `status = "okay";` as the condition required to enable this driver, that's a problem with Zephyr and this driver, because we have started to replace Kconfig flags with `status` property values even though it's breaking backwards compatibility right now (#24745).

I do think that devicetree should follow what Linux does, and the recent update to the devicetree specification to clarify the behavior is justified by the absence of the `status` property as being equivalent to `status = "okay";` in Linux.   However, I agree that this update should be considered a clarification for an edge condition (what to do if it's not there), and that the preferred approach is to provide it.

That here an explicit `status = "disabled";` is being replaced by an implicit `status = "okay";` seems to increase confusion.  Better to make it explicit, IMO.

Reply:
I'm fine with explicitly defining the status field here, but I think the same rule should not be enforced everywhere, specially on the nodes that are essentially mandatory for a valid system configuration. Some examples would include, 
- **Interrupt Controllers** - For most of the systems, there is a single interrupt controller (NVIC for v7-m/v8-m, GIC for v8-a etc) on the hardware and it has to be available in the system configuration. So there is no value for such nodes to provide "status" field. 
- **Firmware nodes** - For heterogeneous systems, there are cases where clocking / pin-muxing / cpu on-off is handled by a separate core (for security purposes) and the kernel has to go through this core (via Secure Software or TI-SCI etc) to get such properties. Again the kernel cannot function properly without such nodes and so they should be explicitly present. 
- And others. 

Overall, I would appreciate if all of this is noted down somewhere in the documentation for future reference like @stephanosio mentioned.. 

I will remove my change-set from here and set status to "okay" in the board-level overlay file..

Reply:
I have no strong opinions here.